R/separate-wider.R

```diff
@@ -0,0 +1,617 @@
+#' Split a string into columns
+#'
+#' @description
+#' `r lifecycle::badge("experimental")`
+#'
+#' Each of these functions takes a string column and splits it into multiple
+#' new columns:
+#'
+#' * `separate_wider_delim()` splits by delimiter.
+#' * `separate_wider_position()` splits at fixed widths.
+#' * `separate_wider_regex()` splits with regular expression matches.
+#'
+#' These functions are equivalent to [separate()] and [extract()], but use
+#' [stringr](https://stringr.tidyverse.org/) as the underlying string
+#' manipulation engine, and their interfaces reflect what we've learned from
+#' [unnest_wider()] and [unnest_longer()].
+#'
+#' @inheritParams unnest_longer
+#' @param cols <[`tidy-select`][tidyr_tidy_select]> Columns to separate.
+#' @param names_sep If supplied, output names will be composed
+#'   of the input column name followed by the separator followed by the
+#'   new column name. Required when `cols` selects multiple columns.
+#'
+#'   For `separate_wider_delim()` you can specify instead of `names`, in which
+#'   case the names will be generated from the source column name, `names_sep`,
+#'   and a numeric suffix.
+#' @param names For `separate_wider_delim()`, a character vector of output
+#'   column names. Use `NA` if there are components that you don't want
+#'   to appear in the output; the number of non-`NA` elements determines the
+#'   number of new columns in the result.
+#' @param delim For `separate_wider_delim()`, a string giving the delimiter
+#'   between values. By default, is interpreted as a fixed string; use
+#'   [stringr::regex()] and friends to split in other ways.
+#' @inheritParams rlang::args_dots_empty
+#' @param too_few What should happen if a value separates into too few
+#'   pieces?
+#'
+#'   * `"error"`, the default, will throw an error.
+#'   * `"debug"` adds additional columns to the output to help you
```

Comment:
Here and in other places where `debug` is an option, I think it would be useful to add a sentence saying something along the lines of "This option is intended to help you debug the issue and address it with one of the other options or further data wrangling. It should not remain in your final code.".